pkg/frontend/clustermanager_get.go

```diff
@@ -20,17 +21,48 @@ func (f *frontend) getClusterManagerConfiguration(w http.ResponseWriter, r *http
 	ctx := r.Context()
 	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
 	vars := mux.Vars(r)
+	badRequestError := fmt.Sprintf("The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
 
-	if f.apis[vars["api-version"]].ClusterManagerConfigurationConverter == nil {
-		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
-		return
+	var (
+		b   []byte
+		err error
+	)
+
+	switch vars["ocmResourceType"] {
+	case "syncset":
+		if f.apis[vars["api-version"]].SyncSetConverter == nil {
+			api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", badRequestError)
+			return
+		}
+
+		b, err = f._getSyncSetConfiguration(ctx, log, r, f.apis[vars["api-version"]].SyncSetConverter)
+	case "machinepool":
+		if f.apis[vars["api-version"]].MachinePoolConverter == nil {
+			api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", badRequestError)
+			return
+		}
+
+		b, err = f._getMachinePoolConfiguration(ctx, log, r, f.apis[vars["api-version"]].MachinePoolConverter)
+	case "syncidentityprovider":
+		if f.apis[vars["api-version"]].SyncIdentityProviderConverter == nil {
+			api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", badRequestError)
+			return
+		}
+
+		b, err = f._getSyncIdentityProviderConfiguration(ctx, log, r, f.apis[vars["api-version"]].SyncIdentityProviderConverter)
+	case "secret":
+		if f.apis[vars["api-version"]].SecretConverter == nil {
+			api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", badRequestError)
+			return
+		}
+
+		b, err = f._getSecretConfiguration(ctx, log, r, f.apis[vars["api-version"]].SecretConverter)
 	}
 
```

Comment:
how about a `default:` clause here that raises an error for unsupported resource type?

Reply:
Done, thanks!